File: GumiPresso_Server/src/main/java/com/ssafy/through/model/service/CommentService.java
package com.ssafy.through.model.service;

import java.util.List;

import com.ssafy.through.model.dto.Comment;

public interface CommentService {

	int insert(Comment comment);

	List<Comment> search(Integer product_id);

	int update(Comment comment);
	
	int delete(Integer id);
	
	Comment select(Integer id);

	int deleteCommentItem(Comment comment);
}

File: GumiPresso_User/app/src/main/java/com/ssafy/gumipresso/model/service/CommentService.kt
package com.ssafy.gumipresso.model.service

import com.ssafy.gumipresso.model.dto.Comment
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Response
import retrofit2.http.*

interface CommentService {
    @GET("/comment/{product_id}")
    suspend fun getComments(@Path("product_id") productId: Int): Response<MutableList<Comment>>

    @POST("/comment/")
    suspend fun insertComment(@Body comment: Comment): Response<MutableList<Comment>>

    @PUT("/comment/")
    suspend fun updateComment(@Body comment: Comment): Response<MutableList<Comment>>

    @POST("/comment/delete")
    suspend fun deleteComment(@Body comment: Comment): Response<MutableList<Comment>>

    @Multipart
    @POST("/comment/image")
    suspend fun insertCommentImage(@Part imageFile: MultipartBody.Part, @Part("comment") product: RequestBody): Response<MutableList<Comment>>

    @Multipart
    @PUT("/comment/image")
    suspend fun updateCommentImage(@Part imageFile: MultipartBody.Part, @Part("comment") product: RequestBody): Response<MutableList<Comment>>
}